content/doc/developer/plugin-governance/deprecating-or-removing-plugin.adoc

```diff
@@ -0,0 +1,37 @@
+---
+layout: developersection
+title: Deprecating or removing a Plugin
+references:
+- url: https://github.com/jenkins-infra/update-center2
+  title: Jenkins Update Center generator
+- url: https://plugins.jenkins.io/ui/search/?labels=deprecated
+  title: List of the deprecated Jenkins plugins
+---
+
+Some plugins may be replaced by others or just become irrelevant (e.g. integration with a service which was shut down).
+We do not recommend deleting source code outright, even a stale or no-longer-relevant code can still be educational.
+However, we do have a mechanism for deprecating or hiding plugins in the Jenkins update centers.
+This page describes the process.
+
+== Deprecating a plugin
+
+. Set a `deprecated` label for the plugin. It can be done in 2 ways:
+** Put a `deprecated` topic in the plugin's GitHub repository.
+   If you have multiple plugins inside a single repository, it will apply to all of them
+** Add a `deprecated` label to the plugin entry in the Update Center's link:https://github.com/jenkins-infra/update-center2/blob/master/src/main/resources/label-definitions.properties[label-definitions.properties] file
+. Update the plugin's documentation to explain the reason of the deprecation
+. Release the plugin and put the deprecation notice into the changelog
```

Comment:
This seems unnecessary and I couldn't find an explanation in the linked mailing list thread. Proposing removal in https://github.com/jenkins-infra/jenkins.io/pull/5786.

Reply:
I think that the removal makes sense, though there have been several cases where a "tombstone release" of a plugin was needed before it was deprecated.  I believe those were very specific cases and don't need to be covered in the general documentation.